### DISCUSSION TITLE
RadioMaster Pocket Radio Controller Programming for NASCAR Wind Tunnel Project

### DISCUSSION THREAD

**Discussion by rwhite77** [EdgeTX/edgetx#5008] [Q&A] [answered]
My name is Roger White. I work for a NASCAR organization. I have been using a digital servo tester to drive two Phoenix Edge 200 amp ESC's controlling two Schubeler DS-51HST fans. We recently purchased a radio master pocket radio controller ELRS and a RadioMaster ER6 2.4GHz ELRS PWM RC Airplane Receiver. I want to replace the servo tester with the ER6 and use the radio to control the speed of the fans remotely. Here is the kicker! I know nothing about programming this radio and it seems there is limited information or I have not looked hard enough. I want to disable all of the buttons except for the rolling pot at the top on the back side. I want to make this active and change the PWM with it. I also want the PWM to be displayed on the screen so the operator knows what speed the fans are running at based on the PWM. Please help!!!

**Comment by pfeerick** [comment, marked as answer]
The first part sounds easy... Delete all the lines from the model settings -> mixes page, and then only assign the back pot (S1) on the Pocket IIRC to the two mix lines that match the channels of the receiver that the ESCs are connected to. 

For displaying the PWM value, I'm not sure offhand... you may be able to set the Radio Settings > PPM Units to 'us', and add the chanel output to a telemetry screen, and shot it that way... will have to check that though. Would require v2.10 of EdgeTx if that works.

**Comment by rwhite77** [comment]
Peter,

Thanks for the reply. Let’s make it happen. We have a test in a few days so I would like to be able to use this!

Thanks again!


Roger White

Team Penske
200 Penske Way
Mooresville, NC 28115

Office: (704) 664-8139
Mobile: (704) 806-4548



________________________________

**Comment by rwhite77** [comment]
Peter,

Just to follow up…you all are going to handle the program? Then all I will have to do is load it? At least that is what I am hoping!

Thanks!


Roger White

Team Penske
200 Penske Way
Mooresville, NC 28115

**Comment by rwhite77** [comment]
Peter,

Do you think this is something you can have done in the next day or two? Worse case, have the setup as discussed but not necessarily have the PWM displayed?

Unfortunately, I am under a little time crunch.

**Reply by pfeerick** [reply]
I've had some time to look at this now, and best I can offer at since time is an issue is this: 

[model10.yml.zip](https://github.com/EdgeTX/edgetx/files/15327722/model10.yml.zip)

Unzip that, and copy the  .yml file to the MODELS folder on the radio's SD card... I'm assuming you only have the stock models on it, so there shouldn't be a `model10.yml` yet... (if there is, just change the number so it doesn't collide with another model number). If you don't know how to select models yet, use the MDL key when at the main screen after powering on, and then use the roller to go up/down the list, and then short press the roller to get the menu with the "select model" option... to well, select the model ;)

I've made the assumption you have the ESCs plugged into CH1 and CH2 of your receiver, so you will either need to match that, or change that on the MIXES screen. And that you are using internal CRSF (ELRS) module. You'll need to configure that to suit, as well as sort out your receiver binding. 

Next, to get your units display (by PWM units, do you mean 1000-2000... i.e. PPM us?)... press the SYS key, use page keys to get to radio setup (page 3), and scroll down (quite a bit) to get to PPM units, and change that (press enter, then use roller to change value) to `us`. Exit out after selecting using RTN. I can't set that bit up for you as it is a radio level setting, and sharing that pre-configured would mean you need to calibrate your radio controls again. 

Now, once you are back at the main view screen, keep pressing one of the page keys to change the main view screen until you get the channels monitor... the us value is shown on the left side, as well as the moving bar. Or, there is another main view screen that has the numbers down the bottom which shows them a bit larger also. 

i.e. 
![image](https://github.com/EdgeTX/edgetx/assets/5500713/2cfbb120-4a4b-4701-8100-29d468493a3c)
or
![image](https://github.com/EdgeTX/edgetx/assets/5500713/68819163-89c2-4aed-9dc5-be70ece91415)

**Reply by rwhite77** [reply]
Peter,

Thanks! I was able to load everything and it all works except for locating the PWM setup. I looked all over page three and it doesn't contain PWM menu item. Any suggestions? I am overlooking it in a sub menu somewhere maybe?

**Reply by rwhite77** [reply]
Peter, 

It appears my time crunch is not as bad now. We will not be using the fans on Monday as expected. I really appreciate your help! Hopefully we can get the PWM bit worked out.

Roger

**Reply by rwhite77** [reply]
Peter,

Another update. I connected everything up, have a flashing green light on the ER6 and the motor does not spin. I only have one channel connected. The display on the radio goes from -100 to 100. I am assuming that is percentage on? Is the problem the units have not been changed? If I plug my servo tester in to the same ESC motor combination, the motor turns on and increases speed with increased pulse width and vise versa with decreased pulse width.

**Reply by rwhite77** [reply]
Peter,

Is the problem that I need to upgrade to v2.10 of EdgeTx as you stated earlier?

**Comment by pfeerick** [comment]
The PPM Units option in radio setup is only available from v2.10 onwards,
yes. Prior to that you needed a custom build flag. So either upgrade to
2.10, or use the cloud build option in EdgeTX Buddy and add the PPM_UNIT
option to us.

-100/100 relates to more to a stick position... i.e. how far (and what
direction) off centre.

**Comment by rwhite77** [comment]
Peter,

I tried connecting the radio by pushing both trim buttons inward and powering up the radio. Then I plugged the USB cable in but it never would connect on the download page.

What am I doing wrong?

Thanks!

Roger White

Team Penske
200 Penske Way
Mooresville, NC 28115

**Comment by pfeerick** [comment]
If you are trying to flash using the USB option, you don't turn the radio
on... Leave it turned on and plug it in (for DFU mode). Bootloader (trims
onwards and poweon) is for accessing the sd card... i.e. using the
save/download firmware option

**Comment by rwhite77** [comment]
Sorry for continuing to be a pain in the butt. I turned the radio off, plugged in the radio via USB and the SD card popped up with all of the contents. But it is still asking me to connect the radio. So I am just a little confused as to why it is not recognizing the radio being plugged in.

Roger White

Team Penske
200 Penske Way
Mooresville, NC 28115

**Reply by pfeerick** [reply]
Not sure why it is coming up if the radio was plugged in when turned off... that should have put it into DFU mode (nothing will be shown on the screen). Anyway, if you are getting the SD card content shown (which means it is not in DFU mode, and the Flash via USB option will not work)... use the `Download .bin` option, save the firmware to the FIRMWARE directory on the SD card. Then you can use the write firmware option in the bootloader to write that firmware. 

![image](https://github.com/EdgeTX/edgetx/assets/5500713/17891faf-3859-4f5d-8174-5684989a642d)

Regarding your receiver

> have a flashing green light on the ER6 

this means you don't have the receiver bound to the transmitter - i.e. the receiver hasn't been told which transmitter to listen to. The light will be solid when it is bound. While it is doing that double blink, go into SYS -> ExpressLRS and scroll down to bind. It should stop blinking, and you can then hold the RTN key to get out of the ExpressLRS Lua script, and then press RTN again to get back to the main view.

**Reply by pfeerick** [reply]
btw, when you do have time/or have an interest to going further than just getting it working ... [manual.edgetx.org](https://manual.edgetx.org/bw-radios) and [www.expresslrs.org/quick-start](https://www.expresslrs.org/quick-start/getting-started/) may help. But lets try and get it there first :grin:

**Comment by rwhite77** [comment]
Peter,

So that worked.

Will I have to do the bind every time we power up? If so, is there anyway around this?

I would like to make it as simple as they turn on the receiver and radio and it’s ready to go.

I will reconnect to my ESC’s in the morning when I am back at work and ensure that the final package works. I have been home for a while. It is 9:37 PM here and I have been going since 5:30 AM this morning. I wanted to get this going.

If I change the owner name to Team Penske will that show up on the screen in place of NASCAR? Or would I need to change the model name?

Thanks again for all of your help! Exactly what we needed!

Roger White

Team Penske
200 Penske Way
Mooresville, NC 28115

**Comment by pfeerick** [comment]
Binding is essentially a one time process... i.e. Like pairing a Bluetooth
speaker with your phone

Where it says NASCAR is the model name, so you need to change that in the
model setup. The owner name is for something else.

**Comment by rwhite77** [comment]
Peter,

Thanks for all your help! Everything seems to work great! I hope I can find some time to look over the manual. Seems like there is a lot you can do with this small device.

How can I leave you positive feedback? Very grateful for you taking the time to make this process easy.

Thanks again!

Roger White

Team Penske
200 Penske Way
Mooresville, NC 28115

**Comment by pfeerick** [comment]
Glad to hear it. Yeah, it is great little radio, and EdgeTX has a lot of
flexibility and options...

Not necessary, thanks is more than enough. Enjoy!

**Comment by rwhite77** [comment]
May I ask what your position is?

Thanks!

Roger White

Team Penske
200 Penske Way
Mooresville, NC 28115
